### DISCUSSION TITLE
Performance advantages / disadvantages

### DISCUSSION THREAD

**Discussion by braebo** [pmndrs/postprocessing#461] [Q&A] [answered]
Hey there!

Have any of the improvement in this library been incorporated into Three?  Or is does this library still out-perform the latest version of Three's builtins?

If so, are there any performance comparisons / benchmarks out there that demonstrate the differences in performance?

I'm asking because I plan on porting a large postprocessing stack that I've built to this library, but it will be quite a lot of work so I would like to weigh the pros and cons beforehand.

Thanks in advance!!

**Comment by vanruesc** [comment, marked as answer]
Hi,

> Have any of the improvement in this library been incorporated into Three?

Passes in `three` have gained consistent `setSize` and `dispose` methods, but other than that, no.

> Or does this library still out-perform the latest version of Three's builtins?

I would say so, yes.

> If so, are there any performance comparisons / benchmarks

https://pmndrs.github.io/postprocessing/public/demo/#performance

Check/uncheck `merge effects` to see the impact on pixel throughput, especially on weaker hardware like iGPUs and mobile devices.

> I'm asking because I plan on porting a large postprocessing stack that I've built to this library

I'm currently working on the [next major release](https://github.com/pmndrs/postprocessing/issues/419) of `postprocessing` which involves some breaking changes, but the concept of passes and effects will largely stay the same. There will be a migration guide for v7 to make the transition easier.

**Reply by braebo** [reply]
Amazing!  I'm incredibly excited for v7 from what I've read so far!  I'll certainly be following it closely 🙏. Thanks!!

**Reply by makc** [reply]
> Three's builtins

they arent builtins, they are examples - normal three.js build does not have any postprocessing
